src/VirtoCommerce.NotificationsModule.Twilio/TwilioSmsNotificationMessageSender.cs

```diff
@@ -23,35 +23,36 @@ public TwilioSmsNotificationMessageSender(IOptions<TwilioSenderOptions> twilioOp
             _smsSendingOptions = smsSendingOptions.Value;
         }
 
-        public virtual bool CanSend(NotificationMessage message)
-        {
-            return message is SmsNotificationMessage;
-        }
+        public virtual bool CanSend(NotificationMessage message) => message is SmsNotificationMessage;
 
         public async Task SendNotificationAsync(NotificationMessage message)
         {
+            ThrowIfSendingNotPossible(message);
+
             TwilioClient.Init(_options.AccountId, _options.AccountPassword);
 
             var smsNotificationMessage = message as SmsNotificationMessage;
 
-            if (smsNotificationMessage == null)
-            {
-                throw new ArgumentNullException(nameof(message));
-            }
-
             try
             {
                 await MessageResource.CreateAsync(
                     body: smsNotificationMessage.Message,
                     from: new PhoneNumber(_smsSendingOptions.SmsDefaultSender),
                     to: smsNotificationMessage.Number
                 );
-
             }
             catch (ApiException ex)
             {
                 throw new SentNotificationException(ex);
             }
         }
+
+        protected virtual void ThrowIfSendingNotPossible(NotificationMessage message)
```

Comment:
We need to suppress this warning in directory.props

Reply:
Did you mean put <NoWarn> in Directory.Build.Props file and revert this code?